shell/src/main/java/alluxio/cli/validation/hdfs/SecureHdfsValidationTask.java

```diff
@@ -39,7 +38,7 @@
    * for more details.
    */
   private static final Pattern PRINCIPAL_PATTERN =
-      Pattern.compile("(?<primary>[\\w][\\w-]*\\$?)(/(?<instance>[\\w]+))?(@(?<realm>[\\w]+))?");
+      Pattern.compile("(?<primary>[^/@]*)(/(?<instance>[^/@]*))?@(?<realm>[^/@]*)");
```

Comment:
Changed to a more lenient format following https://github.com/cloudera/hadoop-common/blob/ca2ff489eb805da4700fb15fa49e539f1c195b89/src/java/org/apache/hadoop/security/KerberosName.java#L53 and the implementation of zookeeper `org.apache.zookeeper.server.auth.KerberosName`.

The previous pattern does not allow dot in the realm like `@ALLUXIO.COM`